pkg/controller/controller.go

```diff
@@ -997,6 +1070,150 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	return err
 }
 
+func (p *csiClaimController) runClaimWorker() {
+	for p.processNextClaimWorkItem() {
+	}
+}
+
+// processNextClaimWorkItem processes items from claimQueue
+func (p *csiClaimController) processNextClaimWorkItem() bool {
+	obj, shutdown := p.claimQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	err := func(obj interface{}) error {
+		defer p.claimQueue.Done(obj)
+		var key string
+		var ok bool
+		if key, ok = obj.(string); !ok {
+			p.claimQueue.Forget(obj)
+			return fmt.Errorf("expected string in workqueue but got %#v", obj)
+		}
+
+		if err := p.syncClaimHandler(key); err != nil {
+			klog.Warningf("Retrying syncing claim %q after %v failures", key, p.claimQueue.NumRequeues(obj))
+			p.claimQueue.AddRateLimited(obj)
```

Comment:
You should not call `Forget` below if you want the obj in the queue.

Reply:
Thanks, didn't notice that!